packages/react/src/components/Toast/index.tsx

```diff
@@ -122,7 +123,7 @@ export default class Toast extends React.Component<ToastProps, ToastState> {
             setRef(toastRef, el);
           }}
           {...defaultProps}
-          {...otherProps}
+          className={defaultProps.className}
```

Comment:
I realize this is a bit unorthodox and we shouldn't really have to do it this way. 

However, in this case, much of the toast's functionality relies upon certain classNames always existing/being set and I don't think consumers of the component will expect adding a custom classname will break all existing functionality/styling (e.g., just trying to add a max-width or some other superficial change). 

```tsx
// It is equivalent to a usage like this:
<Toast show={show} type="caution" className="test" onDismiss={() => setShow(false)}>

// rendering as this
<div tabindex="-1" class="Toast Toast--warning FadeIn--flex FadeIn test">

// instead of this
<div tabindex="-1" class="test">
```

I am definitely open to suggestions on how this could be done differently, though.

Reply:
You're correct in your assumption that `className` should not clobber toast's class name. The cleaner way to implement this might be to destructure `className` from `props` and just spread that as part of default props without setting `className` again. This is primarily how we've implemented it elsewhere, and looks something like this:

```tsx
const { className, ...other } = props
return <div className={className('CauldronClass', classname)} {...other} /> 
```